### ISSUE TITLE
[A11y]'MIT' Link is not descriptive.

### ISSUE THREAD

**Issue by ryuyu** [NuGet/NuGetGallery#7882]
Copied from https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1063680
Steps to Reproduce:
1. Hit the URL: https://www.nuget.org/
2. Navigate to "Sign in" control and hit enter.
3. Navigate to "Sign in with Microsoft" control and hit enter.
4. Search for a key term (say “json”) from search box.
5. Click on "Newtonsoft.json" control and hit enter.
6. Navigate to "Info" and "Statistics" section present on right of the screen.
7. Start the Narrator.
8. Navigate to "MIT" link under info.
9. Observe the Narrator announces the "MIT" link descriptively or not.
 
Actual:
Narrator is not announcing the proper description of 'MIT' link.

Similar issue has been observed with Chrome+NVDA

Expected:
The narrator should announce the description of the link.

Notes:
Context may be lost here that 'MIT' is actually descriptive of the license type. We should add an aria-label indicating "License type: MIT" or something similar.
